New IDE user interface and subfolder (#3150)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/dbt-cloud-ide/ide-glossary.md

---
title: "IDE glossary"
id: ide-glossary
description: "Develop, test, run, and build in the Cloud IDE. With the Cloud IDE, you can compile dbt code into SQL and run it against your database directly"
sidebar_label: IDE glossary
tags: [IDE]
---

The [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) is an interface tool for developers to effortlessly build, test, run, and version-control their dbt projects, and enhance data governance — all from the convenience of your browser. Use the Cloud IDE to compile dbt code into SQL and run it against your database directly -- no command line required!

This page offers comprehensive definitions and terminology of user interface elements, allowing you to navigate the IDE landscape with ease.

## Basic layout

> The IDE is designed to streamline your workflow, and features a popular user interface and layout with all files and folders on the left, editor on the right, and command and console information at the bottom. 

Reviewer: Suggest using the strong verb and simplify:
```suggestion
The IDE streamlines your workflow, and features a popular user interface layout with files and folders on the left, editor on the right, and command and console information at the bottom. 
```

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-basic-layout.jpg" width="100%" title="The Cloud IDE layout includes version control on the upper left, files/folders on the left, editor on the right and command/console at the bottom"/>

> The basic layout of the IDE landing page with the following areas:

Reviewer: ```suggestion
The IDE landing page includes:
```

> 1. **Git repository link &mdash;** Clicking the Git repository link, located on the upper left of the IDE, takes you to your repository on the same active branch. 

Reviewer: Consider using bullets because steps indicate we're conveying an action the user needs to take. UNLESS you have corresponding numbers on the diagram like you do later.

Reviewer: I think the images should come before the explanations. Like you have in "Editing Features" and "Console" sections

Author: gotcha! i am using the numbers to correspond to diagrams - i think because the images were after, it wasn't clear.

    * **Note:** This feature is only available for GitHub or GitLab repositories on multi-tenant dbt Cloud accounts.

2. **Documentation site button &mdash;** Clicking the Documentation site book icon, located next to the Git repository link, leads to the dbt Documentation site. The site is powered by the latest dbt artifacts generated in the IDE using the `dbt docs generate` command from the Command bar.

> 3. [**Version Control &mdash;**](#editing-features) The IDE's powerful Version Control section contains all git-related elements, including the Git actions button and the **Changes** section. 

Reviewer 2: should the Dash be part of the link?

Author: great callout!

> <Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-upper-left.jpg" width="40%" title="The Git repo link, documentation site button, and Version Control menu"/>

Reviewer: Being that the numbers capture this diagram, I suggest moving it up before the numbered paragraphs. You could put a zoom in of the file explorer before step 4.

6. **IDE Status button &mdash;** The IDE Status button, located on the lower right of the IDE, displays the current IDE status. If there is an error in the status or in the dbt code that stops the project from parsing, the button will turn red and display "Error". If there aren't any errors, the button will display a green "Ready" status. To access the [IDE Status modal](#modals-and-menus), simply click on this button.
<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-cmd-status.jpg" width="120%" title="Use the Command bar to write dbt commands and the status button for the current IDE status"/>

## Editing features

The IDE features some delightful tools and layouts to make it easier for you to write dbt code and collaborate with teammates. 

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-editing.jpg" width="100%" title="Use the file editor, format and save button during your development workflow"/>

1. **File Editor &mdash;** The File Editor is where users edit code. Tabs break out the region for each opened file, and unsaved files are marked with a blue dot icon in the tab view.

> 2. **Format button &mdash;** The editor has a **Format** button for `{{ filetypes }}` files. When you click the button, the editor will use `sqlfmt` or `sqlfluff` for SQL files and `black` for Python files to reformat the contents of the file.

Reviewer 2: {{ filetype }} here is meant to be filled in. just remove the first sentence?

4. **Version Control &mdash;** This menu contains all git-related elements, including the Git actions button. The button updates relevant actions based on your editor's state, such as prompting to pull remote changes, commit and sync when reverted commit changes are present, or creating a merge/pull request when appropriate. The dropdown menu on the Git actions button allows users to revert changes, refresh git state, create merge/pull requests, and change branches. You can also [resolve merge conflicts](/docs/collaborate/git/merge-conflicts) and for more info on git, refer to [Version control basics](/docs/collaborate/git/version-control-basics#the-git-button-in-the-cloud-ide).

   - **Version Control Options menu &mdash;** Below the Git Actions button, the **Changes** section, which lists all file changes since the last commit. You can click on a change to open the Git Diff View to see the inline changes. You can also right-click any file and use the file-specific options in the Version Control Options menu.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/version-control-options-menu.png" width="30%" title="Right-click edited files to access Version Control Options menu"/>

> ### Console section

Reviewer: Consider making this and "Additional editing features" level 2 heading. I think nesting headings isn't always necessary.

2. **Compile button &mdash;** The Compile button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code tab.

3. **Build button &mdash;** The build button allows users to quickly access dbt commands related to the active model in the File Editor. The available commands include dbt build, dbt test, and dbt run, with options to include only the current resource, the resource and its upstream dependencies, the resource and its downstream dependencies, or the resource with all dependencies. This menu is available for all executable nodes.

> 4. **Results tab &mdash;** The Results console tab displays the most recent Preview results in tabular format. 

Author: hey @dave-connors-3 just double checking that the data displayed in the image is ok to be shared? is it sample data?

- **Git Diff View &mdash;** Clicking on a file in the **Changes** section of the **Version Control Menu** will open the changed file with Git Diff view. The editor will show the previous version on the left and the in-line changes made on the right.
<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-git-diff-view-with-save.jpg" width="100%" title="The Git Diff View displays the previous version on the left and the changes made on the right of the Editor"/>

- **Markdown Preview console tab &mdash;** The Markdown Preview console tab shows a preview of your .md file's markdown code in your repository, and updates it automatically as you edit your code.
<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-markdown-with-save.jpg" width="100%" title="The Markdown Preview console tab renders markdown code below the Editor tab."/>

- **CSV Preview console tab &mdash;** The CSV Preview console tab displays the data from your CSV file in a table, which updates automatically as you edit the file in your seed directory.
<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-csv.jpg" width="100%" title="View csv code in the CSV Preview console tab below the Editor tab."/>

> ## Invocation history

Reviewer: We might want to define this here as this term is not super obvious. Above, you say " When a command is invoked, the associated logs are shown in the Invocation History Drawer." So maybe a version of that with a little more info?

1. **Invocation History list &mdash;** The left-hand panel of the Invocation History Drawer displays a list of previous invocations in the IDE, including the command, branch name, command status, and elapsed time.

2. **Invocation Summary &mdash;** The Invocation Summary, located above **System Logs**, displays information about a selected command from the Invocation History list , such as the command, its status (`Running` if it's still running), the git branch that was active during the command, and the time the command was invoked.

3. **System Logs toggle &mdash;** The System Logs toggle, located under the Invocation Summary, allows the user to see the full stdout and debug logs for entirety of the invoked command. 

4. **Command Control button &mdash;** Use the Command Control button, located on the right-side, to control your invocation and cancel or rerun a selected run.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-results.jpg" width="100%" title="The Invocation History list displays a list of previous invocations in the IDE"/>

> 5. **Result Status tab &mdash;** Clicking on the Results Status Tabs will filter the Node Status List based on their corresponding status. The available statuses are Pass (successful invocation of a node), Warn (test executed with warning), Error (database error or test failure), Skip (nodes not run due to upstream error), and Queued (nodes that have not executed yet).

Reviewer 2: what do you think about Node Summary tab? or Node Result tab?

Author: i think node summary is good since there is a node result list already. thank you!

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/compiled-code-console-tab.jpg" width="100%" title="Compile results show up in the Compiled Code tab"/>

7. **Lineage tab &mdash;** The Lineage tab in the File Editor displays the active model's lineage or  <Term id="dag" />. By default, it shows two degrees of lineage in both directions (`2+model_name+2`), however you can change it to +model+ (full DAG).
    - Double-click a node in the DAG to open that file in a new tab
    - Expand the DAG and use node selection syntax (select or exclude) to view a subset of your DAG

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/lineage-console-tab.jpg" width="100%" title="View resource lineage in the Lineage tab"/>

## Invocation history
The Invocation History Drawer stores information on dbt invocations in the IDE. When you invoke a command (like execute a dbt command such as `dbt run`), the associated logs are displayed in the Invocation History Drawer. 

You can open the drawer multiple ways: 
- Clicking the `^` icon next to the Command bar on the lower left of the page
- Typing a dbt command and pressing enter

> - Or pressing Control-backtick (or Ctrl + `)

Reviewer 3: i think you need to escape the backtick for it to be rendered properly since backtick has a certain meaning in MD:

https://www.growingwiththeweb.com/2015/06/escaping-backtick-in-markdown.html 

```suggestion
- Or pressing Control-backtick (or Ctrl + \`)
```

Author: good tip- thank you 🙌  
![Screenshot 2023-05-03 at 13 15 02](https://user-images.githubusercontent.com/89008547/235913000-96675e7d-bdd4-437d-8dcd-8b980af4d808.png)
oddly enough it was looking fine without the slash but I've added it

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-console.jpg" width="100%" title="Use the file editor, format, and save button during your development workflow"/>

- **Preview console button &mdash;** When you click on the Preview Button, it runs the SQL in the active file editor regardless of whether you have saved it or not, and sends the results to the Results console tab. To prevent the IDE from returning too much data and causing browser problems, a limit of 500 is automatically added to queries executed via the Preview Button.

- **Compile console button &mdash;** The Compile Button compiles the SQL code from the active File Editor, irrespective of its save status, and outputs it to the Compiled Code console tab.

- **Build console button &mdash;** The build button allows users to quickly access dbt commands related to the active model in the File Editor. The available commands include dbt build, dbt test, and dbt run, with options to include only the current resource, the resource and its upstream dependencies, the resource and its downstream dependencies, or the resource with all dependencies. This menu is available for all executable nodes.

> - **Results console tab &mdash;** The Results console tab displays the most recent Preview results in tabular format. 

Author: hey @dave-connors-3- i'm sure this is dummy data but wanted to triple check that the results here aren't real results? there is a phone num column so wanted to be super sure

Author: hey @dave-connors-3- i'm sure this is dummy data but wanted to triple check that the results here aren't real results? there is a phone num column so wanted to be super sure

sidebar_label: IDE glossary
tags: [IDE]
---

The [dbt Cloud IDE](/docs/cloud/dbt-cloud-ide/develop-in-the-cloud) is an interface tool for developers to effortlessly build, test, run, and version-control their dbt projects - all from the convenience of your browser. Use the Cloud IDE to compile dbt code into SQL and run it against your database directly -- no command line required!

## Landing page

The IDE is designed to streamline your workflow, and features a popular user interface and layout with all files and folders on the left and the editor on the right. 

The basic layout of the IDE landing page is divided into the following areas:

A &mdash; [Git repository link](#git-repository-link)<br />
B &mdash; [Documentation site button](#documentation-site-button)<br />
C &mdash; [Version Control Menu](#version-control-menu)<br />
D &mdash; [File Explorer](#file-explorer)<br />
E &mdash; [Command bar](#command-bar)<br />
F &mdash; [IDE Status button](#ide-status-button)<br />
G &mdash; [Create New File button](#create-new-file-button)<br />

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-landing.jpg" width="100%" title="The Cloud IDE landing page"/>

#### Git Repository link 

> The Git Repository Link displays the name of the active branch. Clicking this element will hyperlink to your repository, pinned to that same branch.

Reviewer 4: This is not true in 100% of cases (In particular, we cannot link to Managed Repositories, I think @StefanDBTLabs would know more about the edgecases here?)

Reviewer 5: It only works on multi tenant for github and gitlab ATM.

#### Documentation site button
The Documentation Site button links to the static dbt Documentation site powered by the most recently generated dbt artifacts in the IDE. This button is clickable once a `dbt docs generate` command is successfully invoked in the Command Bar.  

#### Version Control menu
The Version Control menu contains all git-related elements in the IDE. The Git Actions Button will dynamically update with the relevant action for the state of your editor, including prompt to pull remote changes, commit and sync when reverted commit changes are in the editor, or create a merge/pull request when appropriate. The dropdown menu on the Git Actions Button allows users to revert all changes, refresh git state, create merge/pull requests, and change branches. Refer to [Version control basics](/docs/collaborate/git/version-control-basics#the-git-button-in-the-cloud-ide) for more info. 

Below the Git Actions Button, all changes to files since the last commit will be listed in the **Changes** section. Clicking on any change will open the Git Diff View of the selected file, allowing the user to see the inline changes.

> #### File Explorer 

Reviewer 4: Some of the actions listed below require you are not in _read-only_, which primarily only appears when you're on the `main` branch, or custom configured `main` branch, I wonder if some clarification or link to this `main` behavior in this section could clarify why some options aren't appearing for users who are confused

The File Explorer shows the filetree of your repository. You can:
* Click on any file in the filetree to open the file in the File Editor. 
* Click and drag files between directories to move files. 
* Right click a file to access the sub-menu options like copy file name, copy as `ref`, rename, delete, and more.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-file-explorer-menu.jpg" width="30%" title="Right click files to access sub-menu options"/>

> #### Command bar

Reviewer 4: `Command bar` and `Command Palette` are _very_ similarly named. I wonder if a link in each section which just says "Not to be confused with the Command Palette (link)"

Reviewer 6: i almost feel like calling attention to the other thing could actually invite more confusion than it clears up! 

are there other names for these components you'd prefer?

The Command bar is where [dbt commands](/reference/dbt-commands) are invoked in the IDE. Invoking a command will open the Invocation History Drawer to show the associated logs.

#### IDE Status button
The IDE Status Button reports the current status of the IDE. Anytime there is a status error, or if there is a dbt code error that prevents the project from parsing, this button will turn red and say Error. Clicking on this button brings up the [IDE Status Modal](#IDE-Status-Modal)

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-options-menu.jpg" width="85%" title="Click on the IDE status button to view the options menu"/>

#### Create New File button
This button allows the user to create a new file in their project. 

> ## Editing 

Reviewer 4: You should call out saving here, be it autosave or the big manual save button + save shortcuts

The layout of the IDE editing page is divided into the following areas, and more features.

A &mdash; [File Editor](#file-editor)<br />
B &mdash; [Interaction bar](#interaction-bar)<br />
C &mdash; [Preview button](#preview-button)<br />
D &mdash; [Compile button](#compile-button)<br />
E &mdash; [Build button](#build-button)<br />
F &mdash; [Results subtab](#results-subtab)<br />
G &mdash; [Compiled code subtab](#compiled-code-subtab)<br />
H &mdash; [Lineage subtab](#lineage-subtab)<br />

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-editing.jpg" width="100%" title="The Cloud IDE editing components"/>

#### File Editor
The File Editor is the region where users edit code. This region is broken out by tabs for all files that have been opened. Unsaved Files are marked with a blue dot icon in the tab view. 

The editor has a **Format** button enabled in this area for `{{  filetypes  }}` files, and will use your selected formatter (sqlfmt or sqlfluff) to edit the file contents. 

> 

Reviewer 4: You should call out that this (which I would call the Console) is very contextual! Opening different types of files will provide different Console Tabs and interactions

#### Interaction bar
This interaction bar section sits below the File editor and contains all the sub-tabs and buttons that help you preview, compile, build, view the DAG, and more. 

> Refer to bullets C to H above for more details on the sub-tabs and button details. 

Reviewer 4: Can we provide a link when we say "refer to"?

#### Preview button 
The Preview Button executes the SQL in the active file editor (regardless of save status) against your warehouse, and returns the results to the Results subtab. Queries executed via the Preview Button will automatically have a limit 500 appended to the end to prevent returning excessive data to the IDE and causing browser issues. 

#### Compile button
The Compile Button takes the contents of the active File Editor (regardless of save status) and returns the compiled SQL to the Compiled Code Subtab

#### Build button 

> The build button is a shortcut for the user to invoke dbt commands anchored by the active model in the File Editor. The available commands are dbt build, dbt test, and dbt run with options to include only the current resource, the resource and its upstream dependencies, the resource and its downstream dependencies, or the resource with all dependencies. This menu is enabled for all executable nodes.

Reviewer 4: ```suggestion
The build button is a shortcut for the user to invoke dbt commands anchored by the active model in the File Editor. The available commands are `dbt build`, `dbt test`, and `dbt run` with options to include only the current resource, the resource and its upstream dependencies, the resource and its downstream dependencies, or the resource with all dependencies. This menu is enabled for all executable nodes.
```

#### Results subtab
The Results subtab displays the most recent Preview results in tabular format. 

#### Compiled code subtab
The Compiled Code Tab shows the compiled code generated by the most recent compile invocation triggered by the Compile button.

#### Lineage subtab
The Lineage subtab shows the lineage for the active model in the File Editor. By default, this menu will display 2 degrees of lineage in both directions (`2+model_name+2`). This can be edited to display any desired selection syntax. 

> #### Minimap

Reviewer 4: I find it hard to link these sections to the screenshot above, trying to match letters (A, B, C) which don't exist in the sections here to some which do exist above and inside the screenshot.

I wonder if small screenshots in these individual sections which are just cropped to the relevant content may help disambiguate?

Reviewer 6: i dig it!

A Minimap (code outline) gives you a high-level overview of your source code, which is useful for quick navigation and code understanding. A file's minimap is displayed on the upper-right side of the editor. To quickly jump to different sections of your file, click the shaded area.

> #### Editor Command Palette

Reviewer 4: Do we want to _explicitly_ callout that there's the Monaco command palette and ours just so people know what they're looking at and what makes them different?

Author: hey @heysweet - do you meant specifying this is the dbt Editor Command palette? 

Also in terms of product casing, is it Command Palette or Command palette? Our style guide says we use sentence case, however, UI elements override that. So if the UI name is title case, then the docs need to reflect that.

Reviewer 4: Well in this doc right now we mention _both_ command palettes but each section doesn't reference each other -- that's probably okay, I think @dave-connors-3 thinks its better to avoid calling this out?

I don't have a good answer here for the product casing, I think @bossnunta or @slawrence-work could help with that

> The Editor Command Palette contains text editing shortvuts, and lists the associated keyboard shortcut for each action where applicable. This menu can be accessed using F1 or by right clicking in the File Editor menu and selecting Command Palette.

Reviewer 4: ```suggestion
The Editor Command Palette contains text editing shortcuts, and lists the associated keyboard shortcut for each action where applicable. This menu can be accessed using F1 or by right clicking in the File Editor menu and selecting Command Palette.
```

Hrm I confused myself with some of this language! I saw File Editor and thought File Explorer and was confused because I didn't think you could access the Command Palette from the File Explorer.

I wonder if more links from names of things to sections which explain throughout what they are may be helpful?

Reviewer 4: `F1`? Did we get rid of `cmd+P`? Weren't we also discussing `cmd+shift+P` for Firefox?

Reviewer 6: the `F1` shortcut brings up the `Editor Command Palette` (i.e. monaco) and `cmd+P` brings up the custom React Component one. I would pay a lot of dollars to have only one command palette

Reviewer 6: also removed the reference to File Editor to make it less confusing

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-command-palette.jpg" width="100%" title="Click F1 to access the Editor Command Palette menu for editor shortcuts"/>

#### Editor tab menu
The file tab menu contains helpful options for interacting with open editor tabs. You can access the menu by right-clicking on any open tab. 
<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-tab-menu.jpg" width="100%" title="Right click a tab to view the Editor tab menu options"/>

#### Git Diff View
After clicking on a file in the **Changes** section in the **Version Control Menu**, the changed file will open with the Git Diff view. The editor will display the previous version on the left, and the highlighted, in-line changes made on the right. 

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-git-diff-view.jpg" width="100%" title="The Git Diff View displays the previous version on the left and the changes made on the right of the Editor"/>

> #### Markdown Preview subtab

Reviewer 4: We've called the `Interaction bar` the `Console` and the `subtab` the `consoleTab` in our code, calling the bottom portion the `console`. 

Which do you prefer -- Console and Console Tab or Interaction Bar and Subtab? (I lean console!)

Reviewer 6: I would too! Didn't know the existing verbiage!

Reviewer 2: I like Console and Console Tab actually

### Node result list

The Node result list contains the full list of each Node Result invoked during the dbt run. This can be filtered by clicking on any of the Result Status tabs. 
### Command Control button
Use the command control button to control your invocation and cancel or rerun a selected run.

## Modals and Menus

### File Search
The File Navigation Menu (`CMD/CTRL + O` or the 🔍 icon in the File Explorer) allows for quick searching files by name, and navigating between then with ease.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-file-search.jpg" width="100%" title="The Command History returns a log and detail of all your dbt Cloud invocations."/>

### Global Command Palette
The Global Command Palette contains useful shortcuts for interacting with the IDE, including git actions, specialized dbt commands, as well as compile and preview actions, among others. Use `CMD/CTRL + P` to access the menu.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-global-command-palette.jpg" width="100%" title="The Command History returns a log and detail of all your dbt Cloud invocations."/>

> ### IDE Status modal

Reviewer 4: You should link from here to #### IDE Status Button to say "Here's how you open this!"

The IDE Status modal shows the current error message and debug logs for the server. This also contains an option to restart the IDE. 

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-status-modal.jpg" width="100%" title="The Command History returns a log and detail of all your dbt Cloud invocations."/>

### Commit Changes modal

The Commit Changes modal is accessible via the Git Actions button to commit all changes or via the Version Control Options menu to commit individual changes. The modal allows users to commit and sync the selected changes one a commit message is entered.  

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/commit-changes-modal.png" width="100%" title="The Commit Changes modal is how users commit changes to their branch."/>

### Change Branch modal

The Change Branch modal is how users change git branches in the IDE. This is accessible via the `Change Branch` link above the Version Control Menu, or via the Git Actions Button.

> <Lightbox src="/img/docs/dbt-cloud/cloud-ide/change-branch-modal.png" width="100%" title="The Commit Changes modal is how users change their branch."/>

Reviewer 4: The link to this photo seems broken

The basic layout of the IDE landing page is divided into the following areas:

A &mdash; [Git repository link](#git-repository-link)<br />
B &mdash; [Documentation site button](#documentation-site-button)<br />
C &mdash; [Version Control Menu](#version-control-menu)<br />
D &mdash; [File Explorer](#file-explorer)<br />
E &mdash; [Command bar](#command-bar)<br />
F &mdash; [IDE Status button](#ide-status-button)<br />
G &mdash; [Create New File button](#create-new-file-button)<br />

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-landing.jpg" width="100%" title="The Cloud IDE landing page"/>

#### Git Repository link 
The Git Repository Link displays the name of the active branch. Clicking this element will hyperlink to your repository, pinned to that same branch. Available for Github or Gitlab repos on multi-tenant dbt Cloud. 

#### Documentation site button
The Documentation Site button links to the static dbt Documentation site powered by the most recently generated dbt artifacts in the IDE. This button is clickable once a `dbt docs generate` command is successfully invoked in the Command Bar.  

> #### Version Control menu

Reviewer 2: I feel like this is a "Section" and not a "Menu"

#### Lineage console tab
The Lineage subtab shows the lineage for the active model in the File Editor. By default, this menu will display 2 degrees of lineage in both directions (`2+model_name+2`). This can be edited to display any desired selection syntax. 

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/lineage-console-tab.jpg" width="100%" title="View resource lineage in the Lineage console tab"/>

#### Minimap
A Minimap (code outline) gives you a high-level overview of your source code, which is useful for quick navigation and code understanding. A file's minimap is displayed on the upper-right side of the editor. To quickly jump to different sections of your file, click the shaded area.

#### Editor Command Palette
The Editor Command Palette contains text editing actions, and lists the associated keyboard shortcut for each action where applicable. This menu can be accessed using `F1` or by right-clicking in the text editing area and selecting Command Palette.

<Lightbox src="/img/docs/dbt-cloud/cloud-ide/ide-editor-command-palette-with-save.jpg" width="100%" title="Click F1 to access the Editor Command Palette menu for editor shortcuts"/>

> #### Editor tab menu

Reviewer 2: should this fall in the Modals and Menu section?

A &mdash; [Invocation History list](#invocation-history-list)<br />
B &mdash; [Invocation Summary](#invocation-summary)<br />
C &mdash; [System Logs toggle](#system-logs-toggle)<br />
D &mdash; [Result Status tabs](#result-status-tabs)<br />
E &mdash; [Node result](#node-result)<br />
F &mdash; [Command Control button](#command-control-button)<br />

### Invocation History list

The Invocation History List in the left hand panel of the Invocation History Drawer shows summary details about previous invocations in the IDE. This includes the command, the branch name, the status of the command, and the total elapsed time. 

### Invocation Summary

The Invocation Summary shows the summary details of the selected invocation from the Command History List. This includes the command invoked, the status of the selected command (`Running` if the command is still in progress), the active git branch at the time of the invocation, and the timestamp of the invocation.

### System Logs toggle
The System Logs Toggle allows the user to see the full stdout and debug logs for entirety of the invoked command. 

> ### Result Status tabs

Reviewer 2: Node Status Bar?
Node Status Tab?

@heysweet what is this called in code?

The Results Status Tabs group executed nodes from the command by their result. Clicking on these will filter the Node Status List by the corresponding status. Statuses include:

- Pass - successful invocation of a node
- Warn (tests only) - test executed with warning
- Error - database error when executing node or failure in test
- Skip - nodes that did not run due to upstream error
- Queued - nodes that have yet to execute

### Node result
Each node executed during a dbt command invocation will be summarized in a Node Result toggle, that contains the summary (stdout) and debug logs to easily see the details about each node. The Node Results List contains an entry for every invoked node. 

### Node result list

The Node result list contains the full list of each Node Result invoked during the dbt run. This can be filtered by clicking on any of the Result Status tabs. 

> ### Command Control button

Reviewer 2: general casing for all these headings is inconsistent... are we doing Title Case or Sentence case?

Author: It should be sentence case per our style guide @bossnunta, however UI elements override that. So if the UI name is title case, then the docs need to reflect that.

Use the command control button to control your invocation and cancel or rerun a selected run.

## Modals and Menus

> ### File Search

Reviewer 2: should everything here by h3 or h4?
